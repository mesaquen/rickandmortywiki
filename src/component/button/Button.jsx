import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const primaryColor = props => (props.primary ? props.color : 'white');
const reversePrimaryColor = props => (!props.primary ? props.color : 'white');

const StyledButton = styled.button`
  border: 1px solid ${props => props.color};
  background: ${primaryColor};
  color: ${reversePrimaryColor};
  padding: 10px 5px;
  font-size: 16px;
  cursor: pointer;
  outline: none;
  transition: all 0.2s ease-in-out;

  &:hover {
    background: ${reversePrimaryColor};
    color: ${primaryColor};
  }
`;

export default class Button extends PureComponent {
  static propTypes = {
    color: PropTypes.string,
    primary: PropTypes.bool,
  };

  static defaultProps = {
    primary: false,
    color: 'teal',
  };

  render() {
    const props = this.props;
    return <StyledButton {...props} />;
  }
}
