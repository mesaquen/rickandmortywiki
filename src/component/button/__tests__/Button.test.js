import React from 'react';
import Renderer from 'react-test-renderer';
import Button from '../Button';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-11-16
 */
describe('Button', () => {
  let component;
  let callback = jest.fn();
  component = Renderer.create(<Button onClick={callback}>Sample</Button>);

  test('button is rendererd with default values', function() {
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('buitton click callback is called', () => {
    const instance = component.root.instance;
    instance.handleClick();
    expect(callback.mock.calls).toHaveLength(1);
  });
});
