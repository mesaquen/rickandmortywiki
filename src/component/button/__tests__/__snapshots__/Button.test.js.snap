// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button is rendererd with default values 1`] = `
<button
  className="sc-bdVaJa cUTSOG"
  color="teal"
  onClick={[Function]}
>
  Sample
</button>
`;
